webapp/components/TeaserImage/TeaserImage.vue

```diff
@@ -58,181 +49,142 @@ export default {
       this.$toast.error(file.status, message)
-      setTimeout(() => {
-        this.error = false
-      }, 2000)
-    },
-    transformImage(file) {
-      this.file = file
-      this.showCropper = true
-      this.initEditor()
-      this.initCropper()
-    },
-    initEditor() {
-      this.editor = this.$refs.cropperOverlay
-      this.clearImages()
-      this.thumbnailElement.appendChild(this.editor)
     },
     clearImages() {
-      this.thumbnailElement = document.querySelectorAll('#postdropzone')[0]
-      const thumbnailPreview = document.querySelectorAll('.thumbnail-preview')[0]
-      if (thumbnailPreview) thumbnailPreview.remove()
-      const contributionImage = document.querySelectorAll('.contribution-image')[0]
-      this.oldImage = contributionImage
-      if (contributionImage) contributionImage.remove()
+      const images = document.querySelectorAll('.preview-image')
+      images.forEach((image, index) => {
+        if (index === images.length - 1) image.src = ''
+        else image.remove()
+      })
     },
-    initCropper() {
-      this.image = new Image()
-      this.image.src = URL.createObjectURL(this.file)
-      this.editor.appendChild(this.image)
-      this.cropper = new Cropper(this.image, { zoomable: false, autoCropArea: 0.9 })
+    initCropper(file) {
+      this.showCropper = true
+      this.file = file
+      this.clearImages()
+
+      const imageElement = document.querySelector('#cropping-image')
+      imageElement.src = URL.createObjectURL(file)
+      this.cropper = new Cropper(imageElement, { zoomable: false, autoCropArea: 0.9 })
     },
     cropImage() {
-      this.showCropper = false
+      this.isLoadingImage = true
       const canvas = this.cropper.getCroppedCanvas()
       canvas.toBlob(blob => {
         const imageAspectRatio = canvas.width / canvas.height
-        this.setupPreview(canvas)
-        this.removeCropper()
         const croppedImageFile = new File([blob], this.file.name, { type: this.file.type })
         this.$emit('addTeaserImage', croppedImageFile)
         this.$emit('addImageAspectRatio', imageAspectRatio)
+        this.setupPreview(canvas.toDataURL())
       }, 'image/jpeg')
+
+      this.closeCropper()
     },
-    setupPreview(canvas) {
-      this.image = new Image()
-      this.image.src = canvas.toDataURL()
-      this.image.classList.add('thumbnail-preview')
-      this.thumbnailElement.appendChild(this.image)
+    setupPreview(url) {
+      const previewElement = document.querySelector('.image-uploader .preview-image')
+      previewElement.src = url
+      this.$nextTick((this.isLoadingImage = false))
     },
-    cancelCrop() {
+    closeCropper() {
       this.showCropper = false
-      if (this.oldImage) this.thumbnailElement.appendChild(this.oldImage)
-      this.removeCropper()
-    },
-    removeCropper() {
-      this.editor.removeChild(document.querySelectorAll('.cropper-container')[0])
+      this.cropper.destroy()
```

Comment:
I'm really curious to see this all work! I really like the refactor! and hope it behaves the same.
This is where I'm kicking myself for not adding cypress tests, so that we could make sure it still does.
Anyways, I'll manually test it!

Reply:
Thank you for taking the time to test it! This is really the part where I was most scared to break something...